src/foam/nanos/crunch/ruler/SetUCJStatusOnPut.js

```diff
@@ -0,0 +1,141 @@
+        },
+        {
+          name: 'ucj',
+          type: 'foam.nanos.crunch.UserCapabilityJunction'
+        }
+      ],
+      type: 'CapabilityJunctionStatus',
+      documentation: `
+        Check statuses of all preRequist capabilities - returning:
+        GRANTED: If all pre-reqs are in granted status
+        PENDING: At least one pre-req is still in pending status
+        ACTION_REQUIRED: If not any of the above
+      `,
+      javaCode: `
+        boolean allGranted = true;
+        Capability cap;
+        DAO userCapabilityJunctionDAO = (DAO) x.get("userCapabilityJunctionDAO");
+        List<CapabilityCapabilityJunction> ccJunctions = ( List<CapabilityCapabilityJunction> ) getPrereqs(x, ucj);
+
+        for ( CapabilityCapabilityJunction ccJunction : ccJunctions ) {
+          cap = (Capability) ccJunction.findSourceId(x);
+          if ( ! cap.getEnabled() ) continue;
+          UserCapabilityJunction ucJunction = (UserCapabilityJunction) userCapabilityJunctionDAO.find(AND(
+            EQ(UserCapabilityJunction.SOURCE_ID, ucj.getSourceId()),
+            EQ(UserCapabilityJunction.TARGET_ID, (String) ccJunction.getTargetId())
+          ));
+
+          if ( ucJunction != null && 
+               ( ucJunction.getStatus() == CapabilityJunctionStatus.GRANTED || ucJunction.getStatus() == CapabilityJunctionStatus.GRACE_PERIOD ) 
+             ) 
+              continue;
+          
+
+          if ( ucJunction == null ) {
+            return CapabilityJunctionStatus.ACTION_REQUIRED;
+          }
+          if ( ucJunction.getStatus() != CapabilityJunctionStatus.GRANTED
+               && ucJunction.getStatus() != CapabilityJunctionStatus.PENDING ) {
+            return CapabilityJunctionStatus.ACTION_REQUIRED;
+          }
+          if ( ucJunction.getStatus() == CapabilityJunctionStatus.PENDING ) allGranted = false; 
+        }
+        return allGranted ? CapabilityJunctionStatus.GRANTED : CapabilityJunctionStatus.PENDING;
+      `
+    },
+    {
+      name: 'getPrereqs',
+      args: [
+        {
+          name: 'x',
+          type: 'Context'
+        },
+        {
+          name: 'ucj',
+          type: 'foam.nanos.crunch.UserCapabilityJunction'
+        }
+      ],
+      javaType: 'java.util.List<CapabilityCapabilityJunction>',
+      documentation: `
+        Returns the list of prerequisiteCapabilityJunctions for the target capability of the ucj
+      `, 
+      javaCode: `
+        DAO prerequisiteCapabilityJunctionDAO = (DAO) (x.get("prerequisiteCapabilityJunctionDAO"));
+
+        // get a list of the prerequisite junctions where the current capability is the dependent
+        List<CapabilityCapabilityJunction> ccJunctions = (List<CapabilityCapabilityJunction>) ((ArraySink) prerequisiteCapabilityJunctionDAO
+        .where(EQ(CapabilityCapabilityJunction.SOURCE_ID, ucj.getTargetId()))
+        .select(new ArraySink()))
+        .getArray();
+
+        return ccJunctions;
+      `
+    },
+  ]
+});
```

Comment:
```suggestion
});

```

Reply:
cant commit this one for some reason, added in another commit